crates/typst/src/geom/color.rs

```diff
@@ -144,27 +590,202 @@ impl Color {
+    /// - hue ([`angle`]($angle))
+    /// - saturation ([`ratio`]($ratio))
+    /// - value ([`ratio`]($ratio))
+    /// - alpha ([`ratio`]($ratio))
+    ///
+    /// These components are also available using the [`components`]($color.components)
+    /// method.
+    ///
+    /// ```example
+    /// #square(
+    ///   fill: hsv(30deg, 50%, 60%)
+    /// )
+    /// ```
+    #[func(title = "HSV")]
+    pub fn hsv(
+        /// The real arguments (the other arguments are just for the docs, this
+        /// function is a bit involved, so we parse the arguments manually).
+        args: Args,
+        /// The hue angle.
+        #[external]
+        hue: Angle,
+        /// The saturation component.
+        #[external]
+        saturation: Component,
+        /// The value component.
+        #[external]
+        value: Component,
+        /// The alpha component.
+        #[external]
+        alpha: Component,
+        /// The color to convert to HSL.
+        #[external]
+        color: Color,
+    ) -> SourceResult<Color> {
+        let mut args = args;
+        Ok(if let Some(color) = args.find::<Color>()? {
+            color.to_hsv().into()
+        } else {
+            let h = args.expect("hue component")?;
+            let Component(s) = args.expect("saturation component")?;
+            let Component(v) = args.expect("value component")?;
+            let Component(a) = args.eat()?.unwrap_or(Component(Ratio::one()));
+            HsvColor::new(h, s.get(), v.get(), a.get()).into()
+        })
+    }
+
+    /// Converts this color into its components.
+    ///
+    /// The size and values of this array depends on the color space. You can
+    /// obtain the color space using [`kind`]($color.kind). Below is a table of
+    /// the color spaces and their components:
+    ///
+    /// |       Color space       |     C1    |     C2     |     C3    |   C4   |
+    /// |-------------------------|-----------|------------|-----------|--------|
+    /// | [`luma`]($color.luma)   | Lightness |            |           |        |
+    /// | [`oklab`]($color.oklab) | Lightness |    `a`     |    `b`    |  Alpha |
+    /// | [`linear-rgb`]($color.linear-rgb) | Red  |   Green |    Blue |  Alpha |
+    /// | [`rgb`]($color.rgb)     |    Red    |   Green    |    Blue   |  Alpha |
+    /// | [`cmyk`]($color.cmyk)   |    Cyan   |   Magenta  |   Yellow  |  Key   |
+    /// | [`hsl`]($color.hsl)     |     Hue   | Saturation | Lightness |  Alpha |
+    /// | [`hsv`]($color.hsv)     |     Hue   | Saturation |   Value   |  Alpha |
+    ///
+    /// For the meaning and type of each individual value, see the documentation of
+    /// the corresponding color space. The alpha component is optional and only
+    /// included if the `alpha` argument is `true`. The length of the returned array
+    /// depends on the number of components and whether the alpha component is
+    /// included.
+    ///
+    /// ```example
+    /// #(luma(40%).components() == (40%, ))
```

Comment:
Maybe choose a color with multiple components? I think that'll make for a better example. There's also no need for a comparison because the preview will be next to it, could be something like `#rgb(...).components()`